NuclearData/mgNeutronData/mgNeutronCache_mod.f90

```diff
@@ -0,0 +1,92 @@
+!!
+!! Global Cache for MG Neutron Nuclear Data
+!!
+!! Any active nuclear database writes XSs to this module.
+!! Any material handle reads from this module
+!!
+!! Public Members:
+!!   matCache      -> Array of cached data for materials
+!!   majorantCache -> Array of cached data for majorant XS
+!!
+!! NOTE:
+!!   Cache arrays are deliberatly not targets. This is becouse there should be no pointers to the
+!!   cache. Any update call can change energy of any value so it would not be possible that the
+!!   energy of XSs pointed by pointers would not change silently.
+!!
+module mgNeutronCache_mod
+
+  use numPrecision
+  use genericProcedures,       only : fatalError, numToChar
+  use neutronXsPackages_class, only : neutronMacroXSs
+  use materialHandle_inter,    only : materialHandle
+
+  implicit none
+  private
+
+  !!
+  !! Structure that contains cached data for each MG Neutron Material
+  !!
+  !! Public Members:
+  !!   G_tot  -> Energy group of the total XS in xss
+  !!   G_tail -> Energy group of all XSs in xss except total
+  !!   xss    -> Cached Cross-Section values
+  !!
+  type, public :: cacheMatDat
```

Comment:
Add pointer member to the `Public Members` section? 
We can also say in the doc comment that it is included here because it makes scalability better, perhaps? I imagine that perhaps someone might be confused why it is stored here and not retrieved via the `getMaterial` procedure?